index.bs

```diff
@@ -779,20 +781,24 @@ The <code>length</code> property of the <code>cancel</code> method is <b>1</b>.
 <h4 id="call-readable-stream-pull">CallReadableStreamPull ( stream )</h4>
 
 <ol>
-  <li> If <var>stream</var>@\[[pulling]] is <b>true</b> or <var>stream</var>@\[[draining]] is <b>true</b> or
-    <var>stream</var>@\[[started]] is <b>false</b> or <var>stream</var>@\[[state]] is <code>"closed"</code> or
-    <var>stream</var>@\[[state]] is <code>"errored"</code>, return <b>undefined</b>.
-  <li> Let <var>shouldApplyBackpressure</var> be ShouldReadableStreamApplyBackpressure(<var>stream</var>).
-  <li> If <var>shouldApplyBackpressure</var> is <b>true</b>, return <b>undefined</b>.
-  <li> Set <var>stream</var>@\[[pulling]] to <b>true</b>.
-  <li> Let <var>pullResult</var> be InvokeOrNoop(<var>stream</var>@\[[underlyingSource]], <code>"pull"</code>,
-    (<var>stream</var>@\[[enqueue]], <var>stream</var>@\[[close]], <var>stream</var>@\[[error]]).
-  <li> If <var>pullResult</var> is an abrupt completion,
+  <li> If <var>stream</var>@\[[draining]] is <b>true</b> or <var>stream</var>@\[[started]] is <b>false</b> or
+    <var>stream</var>@\[[state]] is <code>"closed"</code> or <var>stream</var>@\[[state]] is <code>"errored"</code>,
+    return <b>undefined</b>.
+  <li> If <var>stream</var>@\[[pullingPromise]] is not <b>undefined</b>,
     <ol>
-      <li> Call-with-rethrow Call(<var>stream</var>@\[[error]], <b>undefined</b>, «‍<var>pullResult</var>.\[[value]]»).
-      <li> Return <var>pullResult</var>.
+      <li> Upon fulfillment of <var>stream</var>@\[[pullingPromise]], call-with-rethrow
+        CallReadableStreamPull(<var>stream</var>).
```

Comment:
I suggest a design like WritableStreamAdvanceQueue, i.e. set only one fulfillment callback on creation of pullingPromise, and in the callback, check whether CallReadableStreamPull was ever called since the last time we invoked underlyingSource.pull.

CallReadableStreamPull is invoked on any event that may cause shouldApplyBackpressure to be flipped. We shouldn't set a fulfillment callback every time when CallReadableStreamPull is called.